frontend/src/Component/Pages/Roompage.jsx

```diff
@@ -31,8 +33,10 @@ export const RoomPage = () => {
           co2Data[device.dev_eui] = co2Info.data.co2;
         }));
         setCo2Levels(co2Data);
+        setIsLoading(false);  
       } catch (error) {
         console.error('Error fetching devices or CO2 levels:', error);
+        setIsLoading(false);
```

Comment:
Same as the previous comments about using finally rather than multiple setIsLoadings